data/data/bootstrap/files/usr/local/bin/bootkube.sh.template

```diff
@@ -80,6 +78,18 @@ then
     record_service_stage_success
 fi
 
+ICSP_MANIFEST_FILE="$PWD/manifests/image-content-source-policy-0.yaml"
+IDMS_MANIFEST_FILE="$PWD/manifests/image-digest-mirror-set-0.yaml"
+
+if [ -f $ICSP_MANIFEST_FILE ]; then
+then
+    VERSION="$(oc adm release info -o 'jsonpath={.metadata.version}' --icsp-file=$ICSP_MANIFEST_FILE "${RELEASE_IMAGE_DIGEST}")"
+elif [ -f $IDMS_MANIFEST_FILE ]; then
+    # Requires https://issues.redhat.com/browse/OCPNODE-1656
+    VERSION="$(oc adm release info -o 'jsonpath={.metadata.version}' --idms-file=$IDMS_MANIFEST_FILE "${RELEASE_IMAGE_DIGEST}")"
+else
+    VERSION="$(oc adm release info -o 'jsonpath={.metadata.version}' "${RELEASE_IMAGE_DIGEST}")"
+fi
```

Comment:
Perhaps we could simplify with something like?

```shell
MIRROR_FLAG=""
if [ -f $ICSP_MANIFEST_FILE ]; then
     MIRROR_FLAG="--icsp-file=$ICSP_MANIFEST_FILE"
elif [ -f $IDMS_MANIFEST_FILE ]; then
    # Requires https://issues.redhat.com/browse/OCPNODE-1656
      MIRROR_FLAG="--idms-file=$IDMS_MANIFEST_FILE"
fi
    
VERSION="$(oc adm release info -o 'jsonpath={.metadata.version}' "${MIRROR_FLAG}" "${RELEASE_IMAGE_DIGEST}")"
```

This is just to get the idea; there may be some syntax errors in here.

Reply:
Good idea. Updated.